error.go

```diff
@@ -36,29 +38,39 @@ func MakeErrorFromHTTP(resp *http.Response) error {
 
 	var errCode int
 	switch resp.StatusCode {
+	case 400:
+		errCode = ErrorInvalidArgument
 	case 403:
 		errCode = ErrorNotAuthorized
 	case 404:
 		errCode = ErrorNotFound
-	case 400:
-		errCode = ErrorInvalidArgument
+	case 422:
```

Comment:
Hmm, I'd think 409 Conflict fits better for duplicate names.  Thoughts?  Is 422 commonly used for this?